tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/MongoDb/MongoDbIntegration.cs

```diff
@@ -129,6 +89,72 @@ internal static class MongoDbIntegration
             }
 
             return scope;
+
+            static string? GetDatabaseName(object wireProtocol)
+            {
+                if (wireProtocol.TryDuckCast<IWireProtocolWithDatabaseNamespaceStruct>(out var protocolWithDatabaseNamespace)
+                 && protocolWithDatabaseNamespace.DatabaseNamespace is not null
+                 && protocolWithDatabaseNamespace.DatabaseNamespace.TryDuckCast<DatabaseNamespaceStruct>(out var databaseNamespace))
+                {
+                    return databaseNamespace.DatabaseName;
+                }
+
+                return null;
+            }
+
+            static void TryGetHostAndPort(TConnection connection, out string? host, out string? port)
+            {
+                host = null;
+                port = null;
+
+                if (connection.Instance is not null)
+                {
+                    if (connection.EndPoint is IPEndPoint ipEndPoint)
+                    {
+                        host = ipEndPoint.Address.ToString();
+                        port = ipEndPoint.Port.ToString();
+                    }
+                    else if (connection.EndPoint is DnsEndPoint dnsEndPoint)
+                    {
+                        host = dnsEndPoint.Host;
+                        port = dnsEndPoint.Port.ToString();
+                    }
+                }
+            }
+        }
+
+        private static bool TryGetQueryDetails(object wireProtocol, out string? mongoOperationName, out string? collectionName, out string? query)
+        {
+            collectionName = null;
+            query = null;
+            mongoOperationName = null;
+
+            if (wireProtocol.TryDuckCast<IWireProtocolWithCommandStruct>(out var protocolWithCommand)
+             && protocolWithCommand.Command != null
+             && protocolWithCommand.Command.TryDuckCast<IBsonDocumentProxy>(out var bsonDocument))
+            {
+                try
+                {
+                    // the name of the first element in the command BsonDocument will be the operation type (insert, delete, find, etc)
+                    // and its value is the collection name
+                    var firstElement = bsonDocument.GetElement(0);
+                    mongoOperationName = firstElement.Name;
+
+                    if (mongoOperationName is "isMaster" or "hello")
+                    {
+                        return false;
+                    }
+
+                    collectionName = firstElement.Value?.ToString();
+                    query = BsonSerializationHelper.ToShortString(protocolWithCommand.Command);
+                }
+                catch (Exception ex)
+                {
+                    Log.Warning(ex, "Unable to access IWireProtocol.Command properties.");
```

Comment:
It's ok to return true after the exception?

Reply:
Yeah, this is how it was working before - the return false is only if we don't want to create a span at all 👍